oops/backplane/ansa.py

```diff
@@ -284,3 +284,93 @@ def ansa_vertical_resolution(self, event_key):
+def exercise_resolution(bp, printing, saving, dir, refdir,
+                        planet=None, moon=None, ring=None,
+                        undersample=16, use_inventory=False, inventory_border=2):
+    """generic unit tests for orbit.py"""
+
+    ### TODO: use Body.has_rings() once Mark has created it
+    try:
+        if planet is not None:
+            test = bp.ansa_radial_resolution(planet+':ansa')
+            show_info(bp, 'Ansa radial resolution (km)', test,
+                      printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+            test = bp.ansa_vertical_resolution(planet+':ansa')
+            show_info(bp, 'Ansa vertical resolution (km)', test,
+                      printing=printing, saving=saving, dir=dir, refdir=refdir)
+    except:
+        if printing: print("These tests require further development.")
+
+#===============================================================================
+def exercise_geometry(bp, printing, saving, dir, refdir,
+                      planet=None, moon=None, ring=None,
+                      undersample=16, use_inventory=False, inventory_border=2):
+    """generic unit tests for orbit.py"""
+
+    ### TODO: use Body.has_rings() once Mark has created it
+    try:
+        if planet is not None:
+            test = bp.ansa_radius(planet+':ansa')
+            show_info(bp, 'Ansa radius (km)', test,
+                      printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+            test = bp.ansa_altitude(planet+':ansa')
+            show_info(bp, 'Ansa altitude (km)', test,
+                      printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+            test = bp.ansa_longitude(planet+':ansa', 'node')
+            show_info(bp, 'Ansa longitude wrt node (deg)', test*DPR,
+                      printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+            test = bp.ansa_longitude(planet+':ansa', 'aries')
+            show_info(bp, 'Ansa longitude wrt Aries (deg)', test*DPR,
+                      printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+            test = bp.ansa_longitude(planet+':ansa', 'obs')
+            show_info(bp, 'Ansa longitude wrt observer (deg)', test*DPR,
+                      printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+            test = bp.ansa_longitude(planet+':ansa', 'oha')
+            show_info(bp, 'Ansa longitude wrt OHA (deg)', test*DPR,
+                      printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+            test = bp.ansa_longitude(planet+':ansa', 'sun')
+            show_info(bp, 'Ansa longitude wrt Sun (deg)', test*DPR,
+                      printing=printing, saving=saving, dir=dir, refdir=refdir)
+
+            test = bp.ansa_longitude(planet+':ansa', 'sha')
+            show_info(bp, 'Ansa longitude wrt SHA (deg)', test*DPR,
+                      printing=printing, saving=saving, dir=dir, refdir=refdir)
+    except:
```

Comment:
OK, I get it that this is surrounding unit tests, where we might not be able to enumerate all the possible exceptions. This _might_ be a place where an unqualified "except" is OK, but only if (1) you explain this in comments, and (2) if you take some action based on the exception besides ignoring it. Right now, if not printing, you will have no idea that something went wrong. Even if printing, you don't know if this set of tests still needs some development, as the message says, or if something entirely different went wrong. Maybe look into the traceback module. In the except clause, you couls catch the exception, print out a full traceback, and then continue. This way, the user will know exactly wheat went wrong, and exactly where, but it will not stop the unit tests from proceeding.